src/main/java/com/amazon/samplelib/resourceleak/ResourceLeak.java

```diff
@@ -38,5 +42,16 @@ public String testS3(final String bucket, final String key) throws IOException {
         String rawJson = IOUtils.toString(objectData);
         return rawJson;
     }
+    
+    public Object readObjectFromFile(final String objectFile) {
+        try {
+            final ObjectInputStream in = new ObjectInputStream(new BufferedInputStream(new FileInputStream(objectFile)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 48, 49. The resource is closed at line 50. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).